src/components/mc/base/ucc_mc_base.c

```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2020-2021.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "ucc_mc_base.h"
+
+ucc_config_field_t ucc_mc_config_table[] = {
+    {"LOG_LEVEL", "warn",
+     "UCC logging level. Messages with a level higher or equal to the "
+     "selected will be printed.\n"
+     "Possible values are: fatal, error, warn, info, debug, trace, data, func, "
+     "poll.",
+     ucc_offsetof(ucc_mc_config_t, log_component), UCC_CONFIG_TYPE_LOG_COMP},
+
+    {NULL}};
+
+const char *ucc_memory_type_names[] = {
+    [UCC_MEMORY_TYPE_HOST]         = "host",
+    [UCC_MEMORY_TYPE_CUDA]         = "cuda",
+    [UCC_MEMORY_TYPE_CUDA_MANAGED] = "cuda-managed",
+    [UCC_MEMORY_TYPE_ROCM]         = "rocm",
+    [UCC_MEMORY_TYPE_ROCM_MANAGED] = "rocm-managed",
+    [UCC_MEMORY_TYPE_LAST]         = "unknown"};
```

Comment:
We can add the types as we had the components. 

 In the future, you might have x86, aarch64, and DPUs. So for the host type, you might want a more specific identifier.

Reply:
Do you mean rename cpu mem component right now in this pull request?

Reply:
IMO, it might be easier now than later. Is there a reason not to do it now?

Reply:
One reason why I'm not sure about renaming cpu mc to x86-cpu mc is that it's not entirely clear how we will implement architecture specific code. One approach is to keep everything cpu specific in cpu mc so that it will be able to detect and use appropriate implementation. In that case UCC_MEMORY_TYPE_HOST means not specific architecture but runtime system memory. Another approach is to divide cpu mc into separate memory components but in that case we will have a lot of code duplication.